packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroupToolbarItem.tsx

```diff
@@ -19,47 +20,113 @@ export interface ChipGroupToolbarItemProps extends React.HTMLProps<HTMLUListElem
   onClick?: (event: React.MouseEvent) => void;
   /** Aria label for close button */
   closeBtnAriaLabel?: string;
+  /** Position of the tooltip which is displayed if the category name text is longer */
+  tooltipPosition?: 'auto' | 'top' | 'bottom' | 'left' | 'right';
 }
 
-export const ChipGroupToolbarItem: React.FunctionComponent<ChipGroupToolbarItemProps> = ({
-  categoryName = '',
-  children = null,
-  className = '',
-  isClosable = false,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  onClick = (_e: React.MouseEvent) => undefined as any,
-  closeBtnAriaLabel = 'Close chip group',
-  ...props
-}: ChipGroupToolbarItemProps) => {
-  if (React.Children.count(children)) {
-    const renderChipGroup = (id: string, HeadingLevel: any) => (
-      <ul className={css(styles.chipGroup, styles.modifiers.toolbar, className)} {...props}>
-        <li>
-          <HeadingLevel className={css(styles.chipGroupLabel)} id={id}>
-            {categoryName}
-          </HeadingLevel>
-          <ul className={css(styles.chipGroup)}>{children}</ul>
-          {isClosable && (
-            <div className="pf-c-chip-group__close">
-              <ChipButton
-                aria-label={closeBtnAriaLabel}
-                onClick={onClick}
-                id={`remove_group_${id}`}
-                aria-labelledby={`remove_group_${id} ${id}`}
-              >
-                <TimesIcon aria-hidden="true" />
-              </ChipButton>
-            </div>
-          )}
-        </li>
-      </ul>
-    );
+interface ChipGroupToolbarItemState {
+  isTooltipVisible: boolean;
+}
+
+export class ChipGroupToolbarItem extends React.Component<ChipGroupToolbarItemProps, ChipGroupToolbarItemState> {
+  constructor(props: ChipGroupToolbarItemProps) {
+    super(props);
+    this.state = {
+      isTooltipVisible: false
+    };
+  }
+
+  heading = React.createRef<HTMLHeadingElement>();
+
+  static defaultProps: ChipGroupToolbarItemProps = {
+    categoryName: '',
+    children: null,
+    className: '',
+    isClosable: false,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    onClick: (_e: React.MouseEvent) => undefined as any,
+    closeBtnAriaLabel: 'Close chip group',
+    tooltipPosition: 'top'
+  };
 
-    return (
-      <ChipGroupContext.Consumer>
-        {(HeadingLevel: any) => <GenerateId>{randomId => renderChipGroup(randomId, HeadingLevel)}</GenerateId>}
-      </ChipGroupContext.Consumer>
-    );
+  componentDidMount() {
+    this.setState({
+      isTooltipVisible: Boolean(
+        this.heading.current && this.heading.current.offsetWidth < this.heading.current.scrollWidth
+      )
+    });
```

Comment:
Thank you for implementing this tooltip feature. We can definitely use this in Cost Management.

Instead of setting `isTooltipVisible` state, could you test this in the `render`? I'm wondering if `offsetWidth` or `scrollWidth` can change?

It appears that the `render` may not step into the `this.state.isTooltipVisible` block the first time through?

Reply:
Thanks for the feedback Dan.  For this I tried to follow what was already implemented for the Chip.  Do you think the Chip needs to be updated too?

Reply:
Yes, I suspect the `Chip` should be updated as well. It seems problematic to set `isTooltipVisible` once.

Reply:
It seems odd that we need to render twice, but the `ref` needs to be assigned to `HeadingLevel` first. Setting state in `componentDidMount` triggers another call to `render`, which is why it works the second time thru. Let's leave it as is.